File: src/main/java/com/tobe/healthy/notification/repository/NotificationRepositoryCustom.java
package com.tobe.healthy.notification.repository;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.tobe.healthy.notification.presentation.dto.out.NotificationRedDotStatusResult;
import com.tobe.healthy.notification.domain.Notification;
import com.tobe.healthy.notification.domain.NotificationCategory;

public interface NotificationRepositoryCustom {

	Page<Notification> findAllByNotificationType(
		NotificationCategory notificationCategory,
		Long receiverId,
		Pageable pageable);

	List<NotificationRedDotStatusResult> findAllRedDotStatus(
		NotificationCategory notificationCategory,
		Long receiverId);

	boolean findRedDotStatus(Long receiverId);
}

File: src/main/kotlin/com/tobe/healthy/notification/repository/NotificationRepositoryCustom.kt
package com.tobe.healthy.notification.repository

import com.tobe.healthy.notification.domain.dto.out.NotificationRedDotStatusResult
import com.tobe.healthy.notification.domain.entity.Notification
import com.tobe.healthy.notification.domain.entity.NotificationType
import org.springframework.data.domain.Pageable
import org.springframework.data.domain.Slice

interface NotificationRepositoryCustom {
    fun findAllByNotificationType(notificationType: NotificationType, receiverId: Long, pageable: Pageable): Slice<Notification>
    fun findOneById(notificationId: Long, receiverId: Long): Notification?
    fun findAllRedDotStatus(notificationType: NotificationType, receiverId: Long): List<NotificationRedDotStatusResult>
    fun findRedDotStatus(receiverId: Long) : Boolean
}